l2-contracts/contracts/bridge/L2SharedBridge.sol

```diff
@@ -142,42 +151,35 @@ contract L2SharedBridge is IL2SharedBridge, ILegacyL2SharedBridge, Initializable
         bytes32 _assetId,
         bytes memory _l1bridgeMintData
     ) internal pure returns (bytes memory) {
-        // note we use the IL1ERC20Bridge.finalizeWithdrawal function selector to specify the selector for L1<>L2 messages,
+        // note we use the IL1SharedBridge.finalizeWithdrawal function selector to specify the selector for L1<>L2 messages,
         // and we use this interface so that when the switch happened the old messages could be processed
         // solhint-disable-next-line func-named-parameters
         return abi.encodePacked(IL1SharedBridge.finalizeWithdrawal.selector, _assetId, _l1bridgeMintData);
     }
 
-    /// @notice Sets the asset handler address for a given assetId.
-    /// @dev Will be called by ZK Gateway.
-    /// @param _assetId The encoding of the asset on L2.
-    /// @param _assetHandlerAddress The address of the asset handler, which will hold the token of interest.
-    function setAssetHandlerAddress(bytes32 _assetId, address _assetHandlerAddress) external onlyL1Bridge {
-        assetHandlerAddress[_assetId] = _assetHandlerAddress;
-        emit AssetHandlerRegistered(_assetId, _assetHandlerAddress);
-    }
-
     /*//////////////////////////////////////////////////////////////
                             LEGACY FUNCTIONS
     //////////////////////////////////////////////////////////////*/
 
-    /// @notice Finalizes the deposit and mint funds.
+    /// @notice Legacy finalizeDeposit.
+    /// @dev Finalizes the deposit and mint funds.
     /// @param _l1Sender The address of token sender on L1.
     /// @param _l2Receiver The address of token receiver on L2.
     /// @param _l1Token The address of the token transferred.
     /// @param _amount The amount of the token transferred.
-    /// @param erc20Data The ERC20 metadata of the token transferred.
+    /// @param _data The metadata of the token transferred.
     function finalizeDeposit(
         address _l1Sender,
         address _l2Receiver,
         address _l1Token,
         uint256 _amount,
-        bytes calldata erc20Data
+        bytes calldata _data
     ) external override {
-        // onlyBridge {
-        bytes32 assetId = keccak256(abi.encode(L1_CHAIN_ID, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, _l1Token));
+        bytes32 assetId = keccak256(
+            abi.encode(L1_CHAIN_ID, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, bytes32(uint256(uint160(_l1Token))))
```

Comment:
Is the bytes32(uint256(uint160( ))) wrapping required or not? It is sometimes there, sometimes not

Reply:
Maybe this should be moved to the DataEncoding as well?

Reply:
removed it everywhere